adapters/MyTarget/MyTargetAdapter/GADMAdapterMyTargetRewarded.m

```diff
@@ -0,0 +1,174 @@
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+
+	NSUInteger slotId = [GADMAdapterMyTargetUtils slotIdFromCredentials:strongConnector.credentials];
+	guard(slotId > 0) else
+	{
+		MTRGLogError(kGADMAdapterMyTargetErrorSlotId);
+		[strongConnector adapter:self didFailToSetUpRewardBasedVideoAdWithError:[GADMAdapterMyTargetUtils errorWithDescription:kGADMAdapterMyTargetErrorSlotId]];
+		return;
+	}
+
+	_isInterstitialAllowed = NO;
+	_interstitialAd = [[MTRGInterstitialAd alloc] initWithSlotId:slotId];
+	_interstitialAd.delegate = self;
+	[GADMAdapterMyTargetUtils fillCustomParams:_interstitialAd.customParams withConnector:strongConnector];
+	[_interstitialAd.customParams setCustomParam:kMTRGCustomParamsMediationAdmob forKey:kMTRGCustomParamsMediationKey];
+	[strongConnector adapterDidSetUpRewardBasedVideoAd:self];
+}
+
+- (void)requestRewardBasedVideoAd
+{
+	MTRGLogInfo();
+	[_interstitialAd load];
+}
+
+- (void)presentRewardBasedVideoAdWithRootViewController:(UIViewController *)viewController
+{
+	MTRGLogInfo();
+	guard(_isInterstitialAllowed && _interstitialAd) else return;
+	[_interstitialAd showWithController:viewController];
+	_isInterstitialStarted = YES;
+}
+
+- (void)stopBeingDelegate
+{
+	MTRGLogInfo();
+	_connector = nil;
+	if (_interstitialAd)
+	{
+		_interstitialAd.delegate = nil;
+		_interstitialAd = nil;
+	}
+}
+
+#pragma mark - MTRGInterstitialAdDelegate
+
+- (void)onLoadWithInterstitialAd:(MTRGInterstitialAd *)interstitialAd
+{
+	id<GADMRewardBasedVideoAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+	_isInterstitialAllowed = YES;
+	[strongConnector adapterDidReceiveRewardBasedVideoAd:self];
+}
+
+- (void)onNoAdWithReason:(NSString *)reason interstitialAd:(MTRGInterstitialAd *)interstitialAd
+{
+	id<GADMRewardBasedVideoAdNetworkConnector> strongConnector = _connector;
+	NSString *description = [GADMAdapterMyTargetUtils noAdWithReason:reason];
+	MTRGLogError(description);
+	guard(strongConnector) else return;
+	NSError *error = [GADMAdapterMyTargetUtils errorWithDescription:description];
+	[strongConnector adapter:self didFailToLoadRewardBasedVideoAdwithError:error];
+}
+
+- (void)onClickWithInterstitialAd:(MTRGInterstitialAd *)interstitialAd
+{
+	id<GADMRewardBasedVideoAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+	[strongConnector adapterDidGetAdClick:self];
+}
+
+- (void)onCloseWithInterstitialAd:(MTRGInterstitialAd *)interstitialAd
+{
+	id<GADMRewardBasedVideoAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+	[strongConnector adapterDidCloseRewardBasedVideoAd:self];
+}
+
+- (void)onVideoCompleteWithInterstitialAd:(MTRGInterstitialAd *)interstitialAd
+{
+	id<GADMRewardBasedVideoAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+	NSNumber *amount = @0; //must not be nil
```

Comment:
The default reward amount needs to be 1. You could use `[NSDecimalNumber one]` for this.